### PR TITLE
add/trigger_disable_addons Add cancel order with disable addons trigger

### PR DESCRIPTION
**Pull request by vjumpkung**

### CODE REVIEW

File: src/addons/addons.service.ts [hunk 1]
```diff
@@ -34,6 +34,13 @@ export class AddonsService {
       .exec();
   }
 
+  async disableAddon(disableAddonsList: Ref<AddonSchema>[]) {
+    await this.addonModel.updateMany(
+      { _id: { $in: disableAddonsList } },
+      { available: false },
+    );
```
Comment (vjumpkung) [inline src/addons/addons.service.ts]: add `.exec()`

**Review by ishiasaka — CHANGES_REQUESTED** [review]
The cancel addon controller received the addon(s) id for disable only addon that lack

File: src/orders/orders.service.ts [hunk 2]
```diff
@@ -21,6 +23,8 @@ export class OrdersService {
     private readonly orderModel: ReturnModelType<typeof OrdersSchema>,
     @Inject(forwardRef(() => SessionService))
     private readonly sessionService: SessionService,
+    @Inject(AddonsService)
```
Comment (ishiasaka) [inline src/orders/orders.service.ts]: Unnecesary,please Remove it.

Comment (Vjumpkung) [reply src/orders/orders.service.ts:26]: but I use this in my recent function (at line 131) `await this.addonsService.disableAddons(disableAddons);`

Comment (Vjumpkung) [reply src/orders/orders.service.ts:26]: Sorry I am understanding now.

File: src/orders/orders.controller.ts [hunk 3]
```diff
@@ -109,4 +116,22 @@ export class OrdersController {
...
+  })
+  @ApiOperation({
+    summary: 'Cancel order and disable addons',
+  })
+  @HttpCode(HttpStatus.NO_CONTENT)
+  async cancelByAddons(
+    @Param('id', new ParseMongoIdPipe()) id: Types.ObjectId,
+    @Body() addonsList: DisableAddonsDto,
+  ) {
+    await this.ordersService.cancelByAddons(new Types.ObjectId(id), addonsList);
```
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Pipe transformed is to objectId unnecessary to transform in this.

**Review by ishiasaka — CHANGES_REQUESTED** [review]

**Review by ishiasaka — APPROVED** [review]

**Pull request merged**